From 02f4e9d1cc44d8109c85edc2157cd6af415b25d3 Mon Sep 17 00:00:00 2001
From: liutianyu1250 <liutianyu1250@phytium.com.cn>
Date: Thu, 13 Jun 2024 17:50:28 +0800
Subject: [PATCH 090/136] PHYTIUM: net: stmmac: phytium driver add pm support

Test S3 with net device open will got this cut log.
...
   NETDEV WATCHDOG: eth0 (phytium-dwmac): transmit queue 0 timed out 8532ms
   WARNING: CPU: 2 PID: 0 at net/sched/sch_generic.c:525 dev_watchdog+0x234/0x23c
...

Miss pm function in phytium_dwmac_driver.
So add it.

Signed-off-by: liutianyu1250 <liutianyu1250@phytium.com.cn>

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/e4084ab6219ff1e5b2ad5cc68b58be491b3f448f
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
index d31b8e870f64..6e8e44730bec 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
@@ -209,6 +209,7 @@ static struct platform_driver phytium_dwmac_driver = {
 	.remove = phytium_dwmac_remove,
 	.driver = {
 		.name		= "phytium-dwmac",
+		.pm		= &stmmac_pltfr_pm_ops,
 		.of_match_table	= of_match_ptr(phytium_dwmac_of_match),
 		.acpi_match_table = ACPI_PTR(phytium_dwmac_acpi_ids),
 	},
-- 
2.47.0

